File: templates/Uwp/_comp/MVVMLight/Feat.IdentityCommon._VB/ViewModels/UserViewModel.vb
Imports Param_RootNamespace.Helpers
Imports Windows.UI.Xaml.Media.Imaging
Imports GalaSoft.MvvmLight

Namespace ViewModels
    Public Class UserViewModel
        Inherits ViewModelBase

        Private _name As String
        Private _userPrincipalName As String
        Private _photo As BitmapImage

        Public Property Name As String
            Get
                Return _name
            End Get
            Set(value As String)
                [Set](_name, newValue := value)
            End Set
        End Property

        Public Property UserPrincipalName As String
            Get
                Return _userPrincipalName
            End Get
            Set(value As String)
                [Set](_userPrincipalName, newValue := value)
            End Set
        End Property

        Public Property Photo As BitmapImage
            Get
                Return _photo
            End Get
            Set(value As BitmapImage)
                [Set](_photo, value)
            End Set
        End Property

        Public Sub New()
        End Sub
    End Class
End Namespace

File: templates/Wpf/_comp/_shared/Service.IdentityCommon/Param_ProjectName/ViewModels/UserViewModel.cs
using System.Windows.Media.Imaging;

namespace Param_RootNamespace.ViewModels
{
    public class UserViewModel : System.ComponentModel.INotifyPropertyChanged
    {
        private string _name;
        private string _userPrincipalName;
        private BitmapImage _photo;

        public string Name
        {
            get => _name;
            set => Set(ref _name, value);
        }

        public string UserPrincipalName
        {
            get => _userPrincipalName;
            set => Set(ref _userPrincipalName, value);
        }

        public BitmapImage Photo
        {
            get => _photo;
            set => Set(ref _photo, value);
        }

        public UserViewModel()
        {
        }
    }
}
